client.go

```diff
@@ -133,13 +154,16 @@ func (c *Client) Call(params, resp interface{}) error {
 // Any error that c.UnmarshalError or c.Doer returns will not
 // have its cause masked.
 //
-// The request URL field is changed to the actual URL used.
+// If req.URL does not have a host part it will be treated as relative to
+// c.BaseURL.
 func (c *Client) Do(req *http.Request, body io.ReadSeeker, resp interface{}) error {
-	reqURL, err := appendURL(c.BaseURL, req.URL.String())
-	if err != nil {
-		return errgo.Mask(err)
+	var err error
```

Comment:
Perhaps declare this closer to where it's actually needed?
(just after line 175, for example)